iib/workers/tasks/opm_operations.py

```diff
@@ -193,6 +193,134 @@ def _serve_cmd_at_port(serve_cmd, cwd, port, max_tries, wait_time):
+        log.info('Creating empty database file %s', index_db_file)
+        index_db_dir = os.path.dirname(index_db_file)
+        if not os.path.exists(index_db_dir):
+            os.makedirs(index_db_dir, exist_ok=True)
+        with open(index_db_file, 'w'):
+            pass
+
+    return index_db_file
+
+
+def opm_registry_deprecatetruncate(
+    base_dir,
+    index_db,
+    bundles,
+    from_index=None,
+    overwrite_target_index_token=None,
+    container_tool=None,
+):
+    """
+    Deprecate bundles from index.db.
+
+    :param str base_dir: base directory where operation files will be located.
+    :param str index_db: path to index.db used with opm registry deprecatetruncate.
+    :param list bundles: pull specifications of bundles to deprecate.
+    :param str from_index: the pull specification of the container image to parse to determine
+        the registry this token is for.
+    :param str overwrite_target_index_token: the token used for overwriting the input
+        ``from_index`` image. This is required to use ``overwrite_target_index``.
+        The format of the token must be in the format "user:password".
+    :param str container_tool: the container tool to be used to operate on the index image
+    """
+    from iib.workers.tasks.utils import run_cmd, set_registry_token
+
+    log.debug(
+        'Run opm registry deprecatetruncate on database %s and bundles %s',
+        index_db,
+        ' '.join(bundles),
+    )
+
+    cmd = [
+        'opm',
+        'registry',
+        'deprecatetruncate',
+        '--database',
+        index_db,
+        '--bundle-images',
+        ','.join(bundles),
+        '--allow-package-removal',
+    ]
+    if container_tool:
+        cmd.append('--container-tool')
+        cmd.append(container_tool)
+    with set_registry_token(overwrite_target_index_token, from_index):
+        run_cmd(cmd, {'cwd': base_dir}, exc_msg=f'Failed to deprecate the bundles on {index_db}')
+
+
+def deprecate_bundles_fbc(
+    bundles,
+    base_dir,
+    binary_image,
+    from_index,
+    overwrite_target_index_token=None,
+    container_tool=None,
+):
+    """
+    Deprecate the specified bundles from the index image.
+
+    Dockerfile is created only, no build is performed.
+
+    :param list bundles: pull specifications of bundles to deprecate.
+    :param str base_dir: base directory where operation files will be located.
+    :param str binary_image: binary image to be used by the new index image.
+    :param str from_index: index image, from which the bundles will be deprecated.
+    :param str overwrite_target_index_token: the token used for overwriting the input
+        ``from_index`` image. This is required to use ``overwrite_target_index``.
+        The format of the token must be in the format "user:password".
+    :param str container_tool: the container tool to be used to operate on the index image
+    """
+    index_db_file = _get_temp_index_db_fbc(base_dir=base_dir, from_index=from_index)
```

Comment:
Should we use _get_temp_index_db_fbc instead in all add, rm, create functionalities?

Reply:
Yes